core_lib/src/structure/filemanager.cpp

```diff
@@ -292,6 +292,10 @@ Status FileManager::save(Object* object, QString sFileName)
                       tr("Cannot Create Data Directory"),
                       tr("\"%1\" is a file. Please delete the file and try again.").arg(dataInfo.absoluteFilePath()));
     }
+    // There may be old files lying around
+    // from projects before this implementation, cleanup their dirty workfolder
+    // before saving new content
+    removeFilesWithNoXMLReference(sMainXMLFile, sDataFolder);
 
```

Comment:
This is probably not necessary. Current file-saving process will only add used files into the zip archive, so it's ok to leave unused files in temporary working folder and ignore them.

If we really want to clean up the temp working folder,  a case which should be considered is:
1. Delete a sound clip
2. Save project
3. Undo <-- will this undo work?

Reply:
Hmm if that's the case, then i'll probably remove it again

Reply:
Indeed it didn't seem to matter, oh well I've removed it again. You can review now :)